LDMP/data_io.py

```diff
@@ -895,11 +975,130 @@ def __init__(self, parent=None):
         layout.insertWidget(0, self.btnMetadata)
         self.btnMetadata.clicked.connect(self.open_metadata_editor)
 
+        # Output raster file - this should be moved once a job has been
+        # created by calling job_manager.move_job_results().
+        self._output_raster_path = GetTempFilename(".tif")
+
+    def on_region_changed(self, region_info):
+        """
+        Slot raised when the region has changed.
+        """
+        self.validate_overlap()
+
+    def on_input_file_changed(self, success):
+        """
+        Slot raised when a raster or vector file has been uploaded.
+        """
+        if success:
+            self.validate_overlap()
+
+    def validate_overlap(self, user_warning=True) -> bool:
+        """
+        Checks the range within which the geometry of the uploaded file
+        overlaps with that of the selected region. The allowed range should
+        be within the tolerance specified in the settings. Sends a warning
+        that the output will be expanded or clipped if the extents are not
+        the same. Messaging will be sent if the child dialog contains a
+        'msg_bar' attribute of type QMessageBar.
+        """
+        self.msg_bar.clearWidgets()
+
+        region_geom = None
+        if self.region_selector.region_info:
+            region_geom = self.region_selector.region_info.geom
+
+        # No need to go further if the region geometry cannot be determined.
+        if region_geom is None:
+            return False
+
+        file_type = self.input_widget.selected_file_type()
+        if not file_type:
+            return False
+
+        layer_bbox_geom = None
+        source_crs = None
+        if file_type == "raster":
+            lyr = self.input_widget.get_raster_layer()
+            if lyr:
+                layer_bbox_geom = qgis.core.QgsGeometry.fromRect(lyr.extent())
+                source_crs = lyr.crs()
+        else:
+            lyr = self.input_widget.get_vector_layer()
+            if not lyr or not lyr.isValid():
+                return False
+
+            source_crs = lyr.crs()
+
+            # Combine the geometry for all the features
```

Comment:
This is fine for now, but we'll have an issue if countries that span the dateline (like Fiji) try to import land cover data that they have defined in a series of tiles linked with VRTs (as this will merge the polygons for Fiji together, creating a massive polygon from -180 to 180 in WGS84). So we'll in the future need to find a way to use the same login of splitting polygons across the meridian that we do in other tools to allow automatic import of datasets that span the dateline, without creating a dataset that spans the globe.

Reply:
Alright, makes sense to split.